unyt/array.py

```diff
@@ -232,11 +231,7 @@ def _coerce_iterable_units(input_object):
                 raise IterableUnitCoercionError(input_object)
             # This will create a copy of the data in the iterable.
             return unyt_array(input_object)
-        return np.asarray(input_object)
-    else:
-        if isinstance(input_object, Unit):
-            return unyt_quantity(1.0, input_object)
-        return np.asarray(input_object)
+    return np.asarray(input_object)
```

Comment:
I assume this "just works" now without the need for the logic under `else`?

Reply:
Yup, this case is caught by `Unit.__mul__` and `Unit.__rmul__`. The `Unit` class has a higher `__array_priority__` than `unyt_array` so `Unit`'s `__mul__` and `__rmul__` wins over `unyt_array`.